Sources/AsyncHTTPClient/ConnectionPool/HTTPConnectionPool+Waiter.swift

```diff
@@ -0,0 +1,64 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the AsyncHTTPClient open source project
+//
+// Copyright (c) 2021 Apple Inc. and the AsyncHTTPClient project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of AsyncHTTPClient project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+extension HTTPConnectionPool {
+    struct Waiter {
+        struct ID: Hashable {
+            private let objectIdentifier: ObjectIdentifier
+
+            init(_ request: HTTPScheduledRequest) {
+                self.objectIdentifier = ObjectIdentifier(request)
+            }
+        }
+
+        var id: ID {
+            ID(self.request)
+        }
+
+        var request: HTTPScheduledRequest {
+            didSet {
+                self.updateEventLoopRequirement()
+            }
+        }
```

Comment:
Hmm, this feels a bit off. If we can the `request` then the `id` of the waiter also changes. Is that the behaviour we actually want or should it really be called `requestID`?